### PR TITLE
Oriba/add item command

### PR DESCRIPTION
**Pull request by oribarak**
## Description

Add a command for monday.com extension to quickly adding an item to pre-defined board

## Screencast
image

image

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/monday/src/lib/api.tsx [hunk 1]
```diff
@@ -86,6 +86,45 @@ export async function getUser(): Promise<Me> {
   return (result as Record<string, unknown>).me as Me;
 }
 
+export async function getBoardAndUser(): Promise<BoardsResponse> {
```
Comment (freak4pc) [inline extensions/monday/src/lib/api.tsx:89]:
```suggestion
export async function getQuickAddBoardAndUser(): Promise<BoardsResponse> {
```

or, pass the id along to make it a tad more reusable:

```suggestion
export async function getBoardAndUser(boardId: number): Promise<BoardsResponse> {
```

File: extensions/monday/src/addItemToDefaultBoard.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+          error: error as string,
+        }));
+      }
+    }
+    fetch();
+  }, []);
+
+  const board = state.board;
+  if (state.error) {
+    console.log("error view");
```
Comment (freak4pc) [inline extensions/monday/src/addItemToDefaultBoard.tsx:67]:
```suggestion
```

File: extensions/monday/src/addItemToDefaultBoard.tsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    return <ErrorView error={state.error} />;
+  } else if (board) {
+    console.log("theres a board");
```
Comment (freak4pc) [inline extensions/monday/src/addItemToDefaultBoard.tsx:70]:
```suggestion
```

File: extensions/monday/src/addItemToDefaultBoard.tsx [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    return <AddItem board={board} />;
+  } else {
+    console.log("theres no board");
```
Comment (freak4pc) [inline extensions/monday/src/addItemToDefaultBoard.tsx:73]:
```suggestion
```

File: extensions/monday/src/addItemToDefaultBoard.tsx [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    board?: Board;
+    me?: Me;
+    error?: string;
+  }>({ isLoading: true });
+  useEffect(() => {
+    async function fetch() {
+      // Fetch from cache first
+      const [cachedBoard, cachedUser] = await Promise.all([
+        getCachedBoard(),
+        getCachedUser(),
```
Comment (freak4pc) [inline extensions/monday/src/addItemToDefaultBoard.tsx:27]:
```suggestion
        getCachedQuickAddBoard(),
        getCachedUser(),
```

File: extensions/monday/src/addItemToDefaultBoard.tsx [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+import {
+  getCachedUser,
+  getCachedBoard,
+  cacheBoard,
+  cacheUser,
+} from "./lib/persistence";
+import AddItem from "./addItem";
+import { ErrorView } from "./lib/helpers";
+
+export default function AddItemToDefaultBoard() {
```
Comment (freak4pc) [inline extensions/monday/src/addItemToDefaultBoard.tsx:15]: I think calling the feature "Quick-add item" will be nicer and more obvious to people :) WDYT? 

```suggestion
export default function QuickAddItem() {
```

File: extensions/monday/package.json [hunk 7]
```diff
@@ -32,6 +32,26 @@
         "slug",
         "account"
       ]
+    },
+    {
+      "name": "addItemToDefaultBoard",
+      "title": "Add an Item",
```
Comment (freak4pc) [inline extensions/monday/package.json:38]:
```suggestion
      "name": "quickAddItem",
      "title": "Quick-add an Item",
```

File: extensions/monday/package.json [hunk 8]
```diff
@@ -32,6 +32,26 @@
...
+      "subtitle": "monday.com",
+      "description": "Adds an item to a pre-defined board",
+      "mode": "view",
+      "keywords": [
+        "add",
+        "item"
+      ],
+      "preferences": [
+        {
+          "name": "addToBoardId",
```
Comment (freak4pc) [inline extensions/monday/package.json:48]:
```suggestion
          "name": "quickAddBoardId",
```

File: extensions/monday/package.json [hunk 9]
```diff
@@ -32,6 +32,26 @@
         "slug",
         "account"
       ]
+    },
+    {
+      "name": "addItemToDefaultBoard",
```
Comment (freak4pc) [inline extensions/monday/package.json:37]:
```suggestion
      "name": "quickAddItem",
```

File: extensions/monday/package.json [hunk 10]
```diff
@@ -32,6 +32,26 @@
...
+        "add",
+        "item"
+      ],
+      "preferences": [
+        {
+          "name": "addToBoardId",
+          "type": "textfield",
+          "required": true,
+          "title": "Board ID",
+          "description": "ID of the board to use for adding items"
```
Comment (freak4pc) [inline extensions/monday/package.json:52]:
```suggestion
          "description": "ID of the board to quick-add items to"
```

**Review by freak4pc — CHANGES_REQUESTED** [review]
Sooo cool! 🤩 
Thanks for the great PR and feature, this is super useful. 
I left a couple of very minor thoughts, lmk what you think :)

File: extensions/monday/src/addItemToDefaultBoard.tsx (hunk 6 above)
Comment (oribarak) [reply extensions/monday/src/addItemToDefaultBoard.tsx:15]: good idea

**Comment by freak4pc** [comment]
LGTM :) 
Waiting for someone from Raycast team to approve (maybe @pernielsentikaer if he'll have the time :) )

**Review by pernielsentikaer — APPROVED** [review]
Awesome @oribarak 🔥 Let's get it merged! If you want to be added as a contributor, send me your raycast developer username 😄

**Comment by oribarak** [comment]
Thanks for the super quick review and merge :)

My developer username is oribarak, would love to join as a contributor

**Pull request merged**